src/go/rpk/pkg/adminapi/api_transform.go

```diff
@@ -33,6 +37,31 @@ func (a *AdminAPI) DeployWasmTransform(ctx context.Context, t TransformMetadata,
 	return a.sendToLeader(ctx, http.MethodPost, baseTransformEndpoint+"deploy", body, nil)
 }
 
+func validateWasmBinary(file io.Reader) error {
+	// Define the magic bytes for WebAssembly binary module
+	wasmMagicBytes := []byte{0x00, 0x61, 0x73, 0x6D}
+
+	// Not using bufio.Reader because it feels weird to spawn one just to peek
+	buf := make([]byte, 4)
+	if _, err := file.Read(buf); err != nil {
```

Comment:
This needs io.ReadFull

Reply:
is there a reason?

Reply:
Read is not guarantee to complete in one syscall. Although unlikely with only four bytes, it's theoretically possible to be interrupted after reading only two bytes.

https://stackoverflow.com/a/53453140

Write is always all or nothing.

Reply:
that is a pretty solid reason! thanks, will change it